prow/gerrit/adapter/adapter.go

```diff
@@ -248,18 +280,19 @@ func (c *Controller) Sync() {
 
 		now := time.Now()
 		defer func() {
-			gerritMetrics.changeProcessDuration.WithLabelValues(instance).Observe(float64(time.Since(now).Seconds()))
+			gerritMetrics.changeProcessDuration.WithLabelValues(instance, project).Observe(float64(time.Since(now).Seconds()))
 		}()
 
 		// Ignore the error. It is already logged.
-		changes, _ := c.gc.QueryChangesForProject(instance, project, syncTime, c.config().Gerrit.RateLimit)
 		timeQueryChangesForProject := time.Now()
-		log.WithFields(logrus.Fields{"host": instance, "repo": project, "changes": len(changes), "duration(s)": time.Since(now).Seconds()}).Info("Time taken querying for gerrit changes")
 
+		changes, _ := c.gc.QueryChangesForProject(instance, project, syncTime, c.config().Gerrit.RateLimit)
 		if len(changes) == 0 {
 			return
 		}
+		gerritMetrics.gerritRepoQueryLatency.WithLabelValues(instance, project).Observe((float64(time.Since(timeQueryChangesForProject).Seconds())))
```

Comment:
Please move this immediately after the query so that we still log the query time even if the query doesn't return results. We should probably now check if there is an error so that we only log the query time for successful queries, not failures.

Reply:
> Please move this immediately after the query so that we still log the query time even if the query doesn't return results.

Oops, nice catch.

> We should probably now check if there is an error so that we only log the query time for successful queries, not failures.

Ack, will update.